File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/ECommerceWebsite/CategoriesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategoriesRow } from "./CategoriesRow";

export interface CategoriesColumns {
    Id: Column<CategoriesRow>;
    Name: Column<CategoriesRow>;
    Description: Column<CategoriesRow>;
    Photo: Column<CategoriesRow>;
}

export class CategoriesColumns extends ColumnsBase<CategoriesRow> {
    static readonly columnsKey = 'ECommerceWebsite.Categories';
    static readonly Fields = fieldsProxy<CategoriesColumns>();
}

File: AdminPanel/AdminPanel.Web/Modules/ECommerceWebsite/Categories/CategoriesColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel.ECommerceWebsite.Columns;

[ColumnsScript("ECommerceWebsite.Categories")]
[BasedOnRow(typeof(CategoriesRow), CheckNames = true)]
public class CategoriesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public string Photo { get; set; }
}